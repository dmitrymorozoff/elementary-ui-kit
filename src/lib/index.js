import { Button } from "./components/forms/Button";
import { Link } from "./components/forms/Link";
import { Switch } from "./components/forms/Switch";
import { Input } from "./components/forms/Input";
import { Checkbox } from "./components/forms/Checkbox";
import { Card } from "./components/others/Card";
import { Label } from "./components/others/Label";
import { CardButtons } from "./components/others/Card/components/CardButtons";
import { CardContent } from "./components/others/Card/components/CardContent";
import { CardMedia } from "./components/others/Card/components/CardMedia";
import { Collapse } from "./components/others/Collapse";
import { CollapseContent } from "./components/others/Collapse/components/CollapseContent";
import { CollapseTitle } from "./components/others/Collapse/components/CollapseTitle";
import { Modal } from "./components/others/Modal";
import { ModalContent } from "./components/others/Modal/components/ModalContent";
import { ModalTitle } from "./components/others/Modal/components/ModalTitle";
import { ModalActions } from "./components/others/Modal/components/ModalActions";
import { Text } from "./components/others/Text";
import { Search } from "./components/icons/Search";
import { RightArrow } from "./components/icons/RightArrow";
import { Plus } from "./components/icons/Plus";
import { DownArrow } from "./components/icons/DownArrow";
import { UpArrow } from "./components/icons/UpArrow";
import { Checked } from "./components/icons/Checked";
import { Select } from "./components/forms/Select";
import { SelectItem } from "./components/forms/Select/components/SelectItem";

export {
    Button,
    Link,
    Switch,
    Input,
    Checkbox,
    Card,
    CardButtons,
    CardContent,
    CardMedia,
    Text,
    Search,
    RightArrow,
    DownArrow,
    UpArrow,
    Checked,
    Select,
    SelectItem,
    Label,
    Plus,
    Collapse,
    CollapseContent,
    CollapseTitle,
    Modal,
    ModalContent,
    ModalTitle,
    ModalActions,
};
