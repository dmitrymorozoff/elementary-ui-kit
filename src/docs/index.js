import React from "react";
import ReactDOM from "react-dom";
import { Label } from "../lib/components/others/Label";
import { Checkbox } from "../lib/components/forms/Checkbox";
import { Switch } from "../lib";

const App = () => (
    <div>
        <div style={{ margin: "12px 0" }}>
            <Label>Default</Label>
            <Label type="primary">Primary</Label>
            <Label type="success">Success</Label>
            <Label type="warning">Warning</Label>
            <Label type="danger">Danger</Label>
        </div>
        <div>
            <Switch defaultChecked={true} />
        </div>
        <div
            style={{ margin: "12px 0", display: "flex", flexDirection: "column" }}
            onChange={event => {
                // eslint-disable-next-line
                console.log({ name: event.target.name, checked: event.target.checked });
            }}
        >
            <Checkbox label="Item 1" name="item 1" />

            <Checkbox label="Item 2" name="item 2" defaultChecked={true} />

            <Checkbox label="Item 3" name="item 3" disabled={true} />

            <Checkbox label="Item 4" name="item 4" color="#a596e1" />
        </div>
    </div>
);

// eslint-disable-next-line
ReactDOM.render(<App />, document.getElementById("root"));
