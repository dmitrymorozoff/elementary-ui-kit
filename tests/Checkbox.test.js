import React from "react";
import { Checkbox } from "../src/lib";

describe("<Checkbox/>", () => {
    it("should render a default checkbox", () => {
        const component = shallow(<Checkbox />);
        expect(component).toMatchSnapshot();
    });
    it("it checkbox change correctly", () => {
        const mockFunction = jest.fn();
        const component = mount(<Checkbox onChange={mockFunction} />);
        component.find("input").simulate("change", {
            preventDefault() {},
        });
        expect(mockFunction).toHaveBeenCalled();
    });
});
