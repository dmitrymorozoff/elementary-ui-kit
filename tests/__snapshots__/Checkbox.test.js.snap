// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> should render a default checkbox 1`] = `
<styled.label
  label=""
>
  <styled.input
    color="#3a77f8"
    label=""
    onChange={[Function]}
  />
  <styled.span
    color="#3a77f8"
  />
</styled.label>
`;
